lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -195,12 +236,38 @@ def firewall_command(cmdlet_type)
 
           cmd
         end
+
+        # return the current firewall rule settings for icmp_type and protocol are correct
+        # @return [Boolean]
+        def is_set_properly?(icmp_type, protocol)
+          if icmp_type.to_s.empty?
+            return false # rubocop:disable Style/RedundantReturn
+
+          elsif icmp_type.is_a?(Integer)
+            return false unless protocol.start_with?("ICMP")
+            return false unless (0..255).include?(icmp_type)
+
+            true
+
+          elsif icmp_type.is_a?(String)
+            return false if !protocol.start_with?("ICMP") && icmp_type !~ /^\D+$/
+            return false if icmp_type.count("a-zA-Z") > 0 && icmp_type.count(":") > 1
+            return false unless (0..255).include?(icmp_type.to_i)
+
```

Comment:
same  applies here. Instead of this multiple return statements use or for conditions and optimise.